torch_geometric/nn/conv/BayesianGCNConv.py

```diff
@@ -0,0 +1,271 @@
+
+        self.register_buffer('eps_weight',
+                             torch.Tensor(out_channels, in_channels),
+                             persistent=False)
+        self.register_buffer('prior_weight_mu',
+                             torch.Tensor(out_channels, in_channels),
+                             persistent=False)
+        self.register_buffer('prior_weight_sigma',
+                             torch.Tensor(out_channels, in_channels),
+                             persistent=False)
+
+        if bias:
+            self.mu_bias = Parameter(torch.Tensor(out_channels))
+            self.rho_bias = Parameter(torch.Tensor(out_channels))
+            self.register_buffer('eps_bias', torch.Tensor(out_channels),
+                                 persistent=False)
+            self.register_buffer('prior_bias_mu', torch.Tensor(out_channels),
+                                 persistent=False)
+            self.register_buffer('prior_bias_sigma',
+                                 torch.Tensor(out_channels), persistent=False)
+        else:
+            self.register_buffer('prior_bias_mu', None, persistent=False)
+            self.register_buffer('prior_bias_sigma', None, persistent=False)
+            self.register_parameter('mu_bias', None)
+            self.register_parameter('rho_bias', None)
+            self.register_buffer('eps_bias', None, persistent=False)
+
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        self.prior_weight_mu.fill_(self.prior_mean)
+        self.prior_weight_sigma.fill_(self.prior_variance)
+
+        self.mu_weight.data.normal_(mean=self.posterior_mu_init, std=0.1)
+        self.rho_weight.data.normal_(mean=self.posterior_rho_init, std=0.1)
+        if self.mu_bias is not None:
+            self.prior_bias_mu.fill_(self.prior_mean)
+            self.prior_bias_sigma.fill_(self.prior_variance)
+            self.mu_bias.data.normal_(mean=self.posterior_mu_init, std=0.1)
+            self.rho_bias.data.normal_(mean=self.posterior_rho_init, std=0.1)
+        self._cached_edge_index = None
+        self._cached_adj_t = None
+
+    def kl_div(self, mu_q, sigma_q, mu_p, sigma_p):
+        """
+        Calculates kl divergence between two gaussians (Q || P)
+        Parameters:
+             * mu_q: torch.Tensor -> mu parameter of distribution Q
+             * sigma_q: torch.Tensor -> sigma parameter of distribution Q
+             * mu_p: float -> mu parameter of distribution P
+             * sigma_p: float -> sigma parameter of distribution P
+        returns torch.Tensor of shape 0
+        """
+        kl = torch.log(sigma_p) - torch.log(
+            sigma_q) + (sigma_q**2 + (mu_q - mu_p)**2) / (2 *
+                                                          (sigma_p**2)) - 0.5
+        return kl.mean()
+
+    def kl_loss(self):
+        sigma_weight = torch.log1p(torch.exp(self.rho_weight))
+        kl = self.kl_div(self.mu_weight, sigma_weight, self.prior_weight_mu,
```

Comment:
Is it possible to use `torch.nn.functional.kl_div` instead?

Reply:
I would rather keep the kl_div implemented from their version, it seems that they need to use a mu parameter and sigma parameter for the distribution P that we define at the class declaration